pkg/ipamfree/calico.go

```diff
@@ -0,0 +1,29 @@
+package ipamfree
+
+import (
+    "fmt"
+    "log"
+    "net"
+    "os/exec"
+
+    "github.com/nokia/danm/crd/apis/danm/v1"
+    "github.com/nokia/danm/crd/client/clientset/versioned"
+)
+
+func CalicoIpamFreeCondition(danmClient versioned.Interface, ep *v1.DanmEp, dnet *v1.DanmNet) bool {
+    return ep.Spec.NetworkType == "calico"
+}
+
+func CalicoIpamFreeRelease(danmClient versioned.Interface, ep *v1.DanmEp, dnet *v1.DanmNet) error {
```

Comment:
so who will delete the DanmEp in this case?

Reply:
The DanmEp will always be deleted by `danmep.DeleteDanmEp`, after it is successfully called, `ipam.WasIpAllocatedByDanm` will help to determine the need for starting static CNI cleanup process

Reply:
which is the wrong logic, and also one of the reasons I keep pushing the framework with independent handlers
DanmEp deletion and IP cleaning must be tied together - one cannot happen without the other. you can only delete the DanmEp if you know you could clean up the IP. if you could not, you leave the DanmEp and live to see another day

so every cleaner needs to take care of **all** cleaning activities, independently from each other

Reply:
Ahh! This was the missing piece of information! So my assumption was wrong, again. Created an update